webnn/concat.https.any.js

```diff
@@ -0,0 +1,72 @@
+// META: title=test WebNN API concat operation
+// META: global=window,dedicatedworker
+// META: script=./resources/utils.js
+// META: timeout=long
+
+'use strict';
+
+// https://webmachinelearning.github.io/webnn/#api-mlgraphbuilder-concat
+
+const testConcat = async (operandType, syncFlag, inputShapeValues, axis, expectedShapeValue) => {
+  const inputs = [];
+  const namedInputs = {};
+  const TestTypedArray = TypedArrayDict[operandType];
+  for (let i = 0; i < inputShapeValues.length; i++) {
+    inputs.push(builder.input('input' + i, {type: operandType, dimensions: inputShapeValues[i].shape}));
+    namedInputs['input' + i] = new TestTypedArray(inputShapeValues[i].data);
+  }
+  const outputOperand = builder.concat(inputs, axis);
+  const outputs = {'outputOperand': new TestTypedArray(sizeOfShape(expectedShapeValue.shape))};
+  let graph;
+  if (syncFlag) {
+    graph = builder.buildSync({outputOperand});
+    context.computeSync(graph, namedInputs, outputs);
+  } else {
+    graph = await builder.build({outputOperand});
+    await context.compute(graph, namedInputs, outputs);
+  }
+  assert_array_approx_equals_ulp(outputs.outputOperand, expectedShapeValue.data, PrecisionMetrics.concat.ULP[operandType], operandType);
```

Comment:
[minor] Hmm, why does this function name use snake case whereas nearly every other function and variable uses camelCase? (e.g. sizeOfShape, computeSync, loadTestData)?

Reply:
The assertion function `assert_array_approx_equals_ulp()` was intended to add into [`testharness.js`](https://github.com/web-platform-tests/wpt/blob/master/resources/testharness.js) previously, so its' name followed like [`function assert_array_equals()`](https://github.com/web-platform-tests/wpt/blob/master/resources/testharness.js#L1614) of  [`testharness.js`](https://github.com/web-platform-tests/wpt/blob/master/resources/testharness.js). 
Since `assert_array_approx_equals_ulp()` is especial for WebNN API tests, I'm OK to use camelCase style.

Reply:
I see. So there was some external convention.

Reply:
Yes. So what's your suggestion for this assertion function? Thanks.

Reply:
Oh, I generally lean toward internal consistency with whatever code is immediately around it, but there's also the consistency argument of matching other "assert_" functions. I'm fine with whichever you decide - you answered my question either way. ✅

Reply:
I leave this assert to be consistent with other "assert_" functions in testharness.js, thanks.